### PR TITLE
Add server extension using h5grove to replace jupyterlab_hdf

### PR DESCRIPTION
**Pull request by loichuder**
Also:
- Update h5web to 0.0.23 (needed for `H5CoreProvider`, alias to `H5GroveProvider` to match the project renaming)
- H5Web widgets resolve file paths according to notebook location (was JupyterLab server location before)

### CODE REVIEW

File: jupyterlab_h5web/utils.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+def _resolve_path(file_path: Path, base_dir: Path) -> Path:
+    if file_path.is_absolute():
+        return file_path
+
+    return base_dir / file_path
+
+
+def resolve_file_path(
+    file_path: Union[str, Path], base_dir: Union[str, Path]
```
Comment (t20100) [inline jupyterlab_h5web/utils.py:13]: Inverting both parameters would make it easier to read since in the end it does `base_dir/file_path`

Comment (t20100) [reply jupyterlab_h5web/utils.py:13]: or eventually a docstring.

File: jupyterlab_h5web/utils.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+) -> Union[str, Path]:
+    path = file_path if isinstance(file_path, Path) else Path(file_path)
+    base = base_dir if isinstance(base_dir, Path) else Path(base_dir)
+
+    if isinstance(file_path, Path):
+        return str(_resolve_path(path, base))
+
+    return _resolve_path(path, base)
```
Comment (t20100) [inline jupyterlab_h5web/utils.py:21]: Maybe simpler without using `_resolve_path`, something like:

```suggestion
    abs_path = path if path.is_absolute() else (base / path)
    return abs_path if isinstance(abs_path, Path) else str(abs_path)
```

File: jupyterlab_h5web/utils.py (hunk 1 above)
Comment (t20100) [reply jupyterlab_h5web/utils.py:13]: and/or another name making it explicit the returned path is absolute (e.g., `as_absolute_path`)

**Review by t20100 — APPROVED** [review]
LGTM, I left a few comments/suggestions.

A few docstring here and there could help.

File: jupyterlab_h5web/__init__.py [hunk 3]
```diff
@@ -1 +1,24 @@
 from .widget import H5Web
+
+
+def _jupyter_server_extension_points():
+    return [{"module": "jupyterlab_h5web"}]
+
+
+def _load_jupyter_server_extension(server_app):
+    """Registers the API handler to receive HTTP requests from the frontend extension.
```
[the remark this answers is not in the document]
Comment (loichuder) [reply jupyterlab_h5web/__init__.py:9]: I removed the _Parameters_ section

File: jupyterlab_h5web/__init__.py [hunk 4]
```diff
@@ -1 +1,24 @@
...
+    from .handlers import setup_handlers
+
+    setup_handlers(server_app.web_app, server_app.notebook_dir)
+    server_app.log.info(
+        f"Jupyterlab-h5web extension will serve HDF5 files from {server_app.notebook_dir}"
+    )
+
+
+# For backward compatibility with notebook server - useful for Binder/JupyterHub
+load_jupyter_server_extension = _load_jupyter_server_extension
```
[the remark this answers is not in the document]
Comment (loichuder) [reply jupyterlab_h5web/__init__.py:24]: I copied-pasted the cookiecutter for JLab 3 server extensions, the comment is not mine. I assume this is for JLab 2

File: jupyterlab_h5web/utils.py (hunk 2 above)
Comment (loichuder) [reply jupyterlab_h5web/utils.py:21]: In the end, I will enforce the use of `Path`. Less `os` imports and easier to read.

**Review by t20100 — APPROVED** [review]

**Comment by t20100** [comment]
Nice!

**Pull request merged**